tests/tests_pytorch/strategies/test_fsdp.py

```diff
@@ -100,17 +101,21 @@ def _assert_layer_fsdp_instance(self) -> None:
             assert self.layer[layer_num].mixed_precision.buffer_dtype == buffer_dtype
 
 
-class TestFSDPModelAutoWrapped(BoringModel):
-    def __init__(self, wrap_min_params: int = 2):
+class TestBoringModel(BoringModel):
```

Comment:
The boring model only supports automatic optimization. It doesn't implement manual. I suggest dropping the parameterization.

Reply:
Is there any better way to test BoringModel without automatic optimization?

Reply:
Why do you want to test manual optimization? The checkpointing is completely detached from it. At any given point we have a state in the optimizer that we can save or load, how we got to that state doesn't matter. So I suggest to drop the manual optimization and just use the BoringModel as it is.

Reply:
Sorry about the confusion. we are not testing automatic optimization. Instead, we use this function to load the weights while not changing it as shown below:
```python

# Build a new FSDP model, without automatic_optimization
model = TestFSDPModelAutoWrapped(wrap_min_params=wrap_min_params, automatic_optimization=False)

strategy = FSDPStrategy(auto_wrap_policy=partial(size_based_auto_wrap_policy, min_num_params=wrap_min_params))
trainer = Trainer(
    default_root_dir=tmpdir, accelerator="gpu", devices=2, strategy=strategy, precision="16-mixed", max_epochs=1
)

trainer.fit(model, ckpt_path=model_path)
```
Is there a better way to load checkpoint in PyTorch Lightning.

Reply:
`trainer.fit(model, ckpt_path=model_path)` is the only way to load a checkpoint with the full state including trainer state and optimizers.

If loading only the model weights is needed, there is also this option (but doesn't apply here):
```
TestFSDPModelAutoWrapped.load_from_checkpoint(filepath)
```
(and would work only with full state, not with sharded checkpoints)

Reply:
Sounds good, I think this issue is solved?